amp/WEB-INF/src/org/dgfoundation/amp/onepager/util/ActivityUtil.java

```diff
@@ -586,6 +583,52 @@ private static void setActivityStatus(AmpTeamMember ampCurrentMember, boolean dr
         }
     }
 
+    /**
+     * Verifies if the team member can approve an activity from the specified team
+     * See {@link #setActivityStatus(AmpTeamMember, boolean, AmpActivityFields, AmpActivityVersion, boolean, boolean)}
+     * @param atm the team member to check
+     * @param activityTeamId the team id that activity belongs to that the TM can have the approval right
+     * @param oldApprovalStatus the old approval status
+     * @return true if the user is allowed to approve the activity
+     */
+    public static boolean canApprove(AmpTeamMember atm, Long activityTeamId, ApprovalStatus oldApprovalStatus) {
```

Comment:
I checked what `EditActivity` class is doing and yes, it would be good to unify. Besides, it would be good to use validation constants that I defined in this PR instead of plain strings in other logic done by AF. I will report a follow up.

Reply:
great

Reply:
Unfortunately only switching to constants made sense. The code used from `EditActivity` is intended for button text display purpose (e.g. we'll see "edit" for team head, but that team head will automatically approve on edit), while the one from here verifies the team member can be the approver no matter if `edit` or `validate` was displayed to the user.